src/SourceGenerators/Uno.UI.SourceGenerators/DependencyObject/DependencyObjectGenerator.cs

```diff
@@ -108,6 +108,15 @@ private void ProcessType(INamedTypeSymbol typeSymbol)
 
 				if (isDependencyObject && typeSymbol.TypeKind == TypeKind.Class)
 				{
+					if (typeSymbol.Is(_iosViewSymbol))
+					{
+						throw new InvalidOperationException("A 'UIKit.UIView' shouldn't implement 'DependencyObject'. Inherit 'FrameworkElement' instead.");
+					}
+					else if (typeSymbol.Is(_androidViewSymbol))
```

Comment:
@jeromelaban Can you confirm if this is correct?

Your comment https://github.com/unoplatform/uno/issues/6758#issuecomment-898544729 was specifically about ViewGroup type. But I thought the same issue should be with the base Android.Views.View type.

Reply:
ViewGroup or View is fine. Though the validation should not exclude types also inheriting from UIElement.

Reply:
`isDependencyObject` will be false for UIElement because of the second check:

```
(typeSymbol.BaseType?.GetAllInterfaces().None(t => SymbolEqualityComparer.Default.Equals(t, _dependencyObjectSymbol)) ?? true
```